src/app/timer-form/timer-form.component.ts

```diff
@@ -0,0 +1,77 @@
+import { Component } from '@angular/core';
+import {
+  AbstractControl,
+  FormControl,
+  FormGroup,
+  ValidationErrors,
+  ValidatorFn,
+} from '@angular/forms';
+import { Timer, TimerService } from '../service/timer-service.service';
+import { Observable } from 'rxjs';
+
+@Component({
+  selector: 'app-timer-form',
+  templateUrl: './timer-form.component.html',
+  styleUrls: ['./timer-form.component.scss'],
+})
+export class TimerFormComponent {
+  timerForm!: FormGroup;
+  timers$!: Observable<Timer[]>;
+
+  constructor(private timerService: TimerService) {
+    this.timerForm = new FormGroup(
+      {
+        label: new FormControl(null),
+        hours: new FormControl(null, [
+          this.typeValidator(),
+          this.minValue(),
```

Comment:
Why not build in NG validators?

Reply:
min, max replaced with built-in validators